src/DbPostgreSQLAdapter/Field.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include "DbAdapterInterface/IField.h"
+#include "DbAdapterInterface/IBinaryValue.h"
+
+namespace systelab::db::postgresql {
+
+	class Field : public IField
+	{
+	public:
+		Field(const unsigned int index, const std::string& name, const FieldTypes type, const std::string& defaultValue, const bool primaryKey);
+		~Field() override = default;
+
+		unsigned int getIndex() const override;
+		std::string getName() const override;
+		FieldTypes getType() const override;
+		bool isPrimaryKey() const override;
+
+		bool hasNullDefaultValue() const override;
+		bool getBooleanDefaultValue() const override;
+		int getIntDefaultValue() const override;
+		double getDoubleDefaultValue() const override;
+		std::string getStringDefaultValue() const override;
+		std::chrono::system_clock::time_point getDateTimeDefaultValue() const override;
+		IBinaryValue& getBinaryDefaultValue() const override;
+
+	private:
+		unsigned int m_index;
+		std::string m_name;
+		FieldTypes m_type;
+		bool m_primaryKey;
+
+		bool m_nullDefaultValue;
```

Comment:
**Not for now** but we should evaluate if all this fuzz could be improved using `std::variant` (https://en.cppreference.com/w/cpp/utility/variant)

Reply:
Yep, the is something here that gives itches